zio/csvio.(*preprocess).parseField panic

```
$ git describe --dirty --tags
v0.30.0-5-g5c4868de-dirty
$ go run ./cmd/zq -i csv - <<<'""'
panic: runtime error: index out of range [-1]
goroutine 35 [running]:
runtime/debug.Stack(0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.16.5/libexec/src/runtime/debug/stack.go:24 +0x9f
github.com/brimdata/zed/zqe.RecoverError(0x1867020, 0xc000140ae0, 0xc000140ae0, 0x0)
	/Users/noah/looky/zed/zqe/error.go:163 +0x26
github.com/brimdata/zed/driver.safePull.func1(0xc0000bbf78)
	/Users/noah/looky/zed/driver/driver.go:193 +0x4b
panic(0x1867020, 0xc000140ae0)
	/usr/local/Cellar/go/1.16.5/libexec/src/runtime/panic.go:965 +0x1b9
github.com/brimdata/zed/zio/csvio.(*preprocess).parseField(0xc00013cb80, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/noah/looky/zed/zio/csvio/preprocess.go:82 +0x685
github.com/brimdata/zed/zio/csvio.(*preprocess).Read(0xc00013cb80, 0xc0003ab000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/Users/noah/looky/zed/zio/csvio/preprocess.go:37 +0x129
bufio.(*Reader).fill(0xc000144de0)
	/usr/local/Cellar/go/1.16.5/libexec/src/bufio/bufio.go:101 +0x108
bufio.(*Reader).ReadSlice(0xc000144de0, 0xa, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.16.5/libexec/src/bufio/bufio.go:360 +0x3d
encoding/csv.(*Reader).readLine(0xc00015a990, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/Cellar/go/1.16.5/libexec/src/encoding/csv/reader.go:218 +0x49
encoding/csv.(*Reader).readRecord(0xc00015a990, 0x0, 0x0, 0x0, 0x0, 0x0, 0x203000, 0x203000, 0x203000)
	/usr/local/Cellar/go/1.16.5/libexec/src/encoding/csv/reader.go:266 +0xf2
encoding/csv.(*Reader).Read(0xc00015a990, 0x0, 0x0, 0xc00011aca0, 0x100f0bb, 0xc000159c00)
	/usr/local/Cellar/go/1.16.5/libexec/src/encoding/csv/reader.go:187 +0x57
github.com/brimdata/zed/zio/csvio.(*Reader).Read(0xc000153360, 0x1f925b8, 0x380, 0xc000159c00)
	/Users/noah/looky/zed/zio/csvio/reader.go:45 +0x32
github.com/brimdata/zed/zbuf.(*scanner).Read(0xc000144f60, 0x0, 0x64, 0xc000159c00)
	/Users/noah/looky/zed/zbuf/scanner.go:111 +0x65
github.com/brimdata/zed/zbuf.readBatch(0x1a25c00, 0xc000144f60, 0x64, 0x0, 0x0, 0x0, 0x0)
	/Users/noah/looky/zed/zbuf/batch.go:41 +0xa3
github.com/brimdata/zed/zbuf.(*puller).Pull(0xc0001259a0, 0x0, 0x0, 0x0, 0x0)
	/Users/noah/looky/zed/zbuf/batch.go:86 +0x4e
github.com/brimdata/zed/zbuf.(*namedScanner).Pull(0xc0001259c0, 0x0, 0x0, 0x0, 0x0)
	/Users/noah/looky/zed/zbuf/scanner.go:152 +0x3a
github.com/brimdata/zed/proc/from.(*Proc).Pull(0xc00013ccc0, 0x0, 0x0, 0x0, 0x0)
	/Users/noah/looky/zed/proc/from/from.go:45 +0x4d
github.com/brimdata/zed/driver.safePull(0x253aaa8, 0xc00013ccc0, 0x0, 0x0, 0x0, 0x0)
	/Users/noah/looky/zed/driver/driver.go:195 +0x6a
github.com/brimdata/zed/driver.run.func1(0x253aaa8, 0xc00013ccc0, 0xc0003b0000)
	/Users/noah/looky/zed/driver/driver.go:108 +0x39
created by github.com/brimdata/zed/driver.run
	/Users/noah/looky/zed/driver/driver.go:106 +0x134


exit status 1
```

--- a/zio/csvio/preprocess.go
+++ b/zio/csvio/preprocess.go
@@ -79,7 +79,7 @@ func (p *preprocess) parseField() ([]byte, error) {
 			ending := []byte{c}
 			if hasstr {
 				// If field had quotes wrap entire field in quotes.
-				if last := len(p.scratch) - 1; p.scratch[last] == '\r' {
+				if last := len(p.scratch) - 1; last > 0 && p.scratch[last] == '\r' {
 					ending = []byte("\r\n")
 					p.scratch = p.scratch[:last]
 				}

--- a/zio/csvio/preprocess_test.go
+++ b/zio/csvio/preprocess_test.go
@@ -12,12 +12,18 @@ import (
 
 func TestPreprocess(t *testing.T) {
 	const input = `
+,,
+"","",""
+"""","""",""""
 field1,"field"2,field"3" my friend
 field4,"field"5 with "multiple" quotes "to" escape,field6
 """,""",""" has a couple "" embedded quotes and a , comma",""" """
 x,"hello,
 "" world , " foo,y`
 	const expected = `
+,,
+"","",""
+"""","""",""""
 field1,"field2","field3 my friend"
 field4,"field5 with multiple quotes to escape",field6
 """,""",""" has a couple "" embedded quotes and a , comma",""" """

